lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+/// <reference types="@panva/jose" />
+/// <reference types="node" />
+
+/**
+ * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md
+ */
+declare module 'openid-client' {
+  import { JWKS } from '@panva/jose'
+  import { IncomingMessage } from 'http'
+  import { Http2ServerRequest } from 'http2'
+
+  /**
+   * @see https://medium.com/@darutk/diagrams-of-all-the-openid-connect-flows-6968e3990660
+   */
+  enum AuthResponseType {
+    CODE = 'code',
+    TOKEN = 'token',
+    ID_TOKEN = 'id_token',
+  }
+
+  /**
+   * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md#client-authentication-methods
+   */
+  enum TokenAuthMethod {
+    BASIC = 'client_secret_basic',
+    POST = 'client_secret_post',
+    CLIENT_SECRET_JWT = 'client_secret_jwt',
+    PRIVATE_KEY_JWT = 'private_key_jwt',
+    TLS_CLIENT_AUTH = 'tls_client_auth'
+  }
+
+  /**
+   * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md#new-clientmetadata-jwks
+   */
+  export interface IClientMetadata {
+    client_id: string
+    client_secret?: string
+    id_token_signed_response_alg?: string
+    id_token_encrypted_response_alg?: string
+    id_token_encrypted_response_enc?: string
+    userinfo_signed_response_alg?: string
+    userinfo_encrypted_response_alg?: string
+    userinfo_encrypted_response_enc?: string
+    redirect_uris?: string[]
+    response_types?: AuthResponseType[]
+    post_logout_redirect_uris?: string[]
+    default_max_age?: number
+    require_auth_time?: boolean
+    request_object_signing_alg?: string
+    request_object_encryption_alg?: string
+    request_object_encryption_enc?: string
+    token_endpoint_auth_method?: TokenAuthMethod
+    introspection_endpoint_auth_method?: TokenAuthMethod
+    revocation_endpoint_auth_method?: TokenAuthMethod
+    token_endpoint_auth_signing_alg?: string
+    introspection_endpoint_auth_signing_alg?: string
+    revocation_endpoint_auth_signing_alg?: string
+    tls_client_certificate_bound_access_tokens?: boolean
+
+    [key: string]: any
```

Comment:
`unknown` is a safer type to use here, so the developer has to explicitly cast it to what they believe the type should be